terraform/azure/main.tf

```diff
@@ -153,10 +146,6 @@ resource "azurerm_kubernetes_cluster" "jupyterhub" {
       "hub.jupyter.org/node-purpose" = "core",
       "k8s.dask.org/node-purpose"    = "core"
     }, var.node_pools["core"][0].labels)
-    # node_taints field will be removed in v4.0 of the Azure Provider since the AKS API
-    # doesn't allow arbitrary node taints on the default node pool. A warning that
-    # this field will be deprecated is produced.
-    node_taints = concat([], var.node_pools["core"][0].taints)
```

Comment:
This is removed for `default_node_pool`, not for all node pools - so its just removing whats being deprecated and wasn't set.